src/types.rs

```diff
@@ -227,3 +230,56 @@ impl SignatureVerification {
         self.inner.trusted()
     }
 }
+
+/// The result of a call to {@link OlmMachine.importExportedRoomKeys} or
+/// {@link OlmMachine.importBackedUpRoomKeys}.
+#[derive(Clone, Debug)]
+#[wasm_bindgen]
+pub struct RoomKeyImportResult {
+    /// The number of room keys that were imported.
+    #[wasm_bindgen(readonly, js_name = "importedCount")]
+    pub imported_count: usize,
+
+    /// The total number of room keys that were found in the export.
+    #[wasm_bindgen(readonly, js_name = "totalCount")]
+    pub total_count: usize,
+
+    /// The map of keys that were imported.
+    ///
+    /// A map from room id to a map of the sender key to a set of session ids.
+    keys: BTreeMap<OwnedRoomId, BTreeMap<String, BTreeSet<String>>>,
+}
+
+#[wasm_bindgen]
+impl RoomKeyImportResult {
+    /// The keys that were imported.
+    ///
+    /// A Map from room id to a Map of the sender key to a Set of session ids.
+    ///
+    /// Typescript type: `Map<string, Map<string, Set<string>>`.
+    pub fn keys(&self) -> Map {
+        let key_map = Map::new();
+
+        for (room_id, room_result) in self.keys.iter() {
+            let room_map = Map::new();
+            key_map.set(&JsString::from(room_id.to_string()), &room_map);
+
+            for (sender_key, sessions) in room_result.iter() {
+                let s: Array = sessions.iter().map(String::as_ref).map(JsString::from).collect();
+                room_map.set(&JsString::from(sender_key.as_ref()), &Set::new(&s));
```

Comment:
Does the wasm binding for `Set::new` accept an iterator instead of an allocated collection? Mostly curious here (I think in JS `Set.new` will take any argument that implements the iteration protocol, e.g. a generator, and insert each value it yields into the set).

Reply:
short answer: [not easily](https://github.com/rustwasm/wasm-bindgen/issues/1478), as far as I know.